Improve fork check in Git hooks installation

willy-liu opened the pull request:
This change allows users to modify the repository name when working with a fork.

Modifications:
- Introduced repository name to store the repository name dynamically.
- Replaced fixed occurrences of lab0-c in API calls with current repository name.
- Improved fork validation by:
  1. Checking if the repository is a fork.
  2. Verifying that the parent repository's full name.

Note:
- This change requires a tool to parse the response from github.com.

willy-liu commented:
1. Correct the regular expressions for `ACCOUNT` and `REPO_NAME`.
2. Remove checking the command `jq`
3. Rewrite jq code with sed/awk.

jserv commented on scripts/install-git-hooks:
```diff
@@ -31,7 +31,8 @@ fi
 
-ACCOUNT=$(git config -l | grep -w remote.origin.url | sed -e 's/^.*github.com[\/:]\(.*\)\/lab0-c.*/\1/')
+ACCOUNT=$(git config --get remote.origin.url | awk -F'[:/]' '{print $(NF-1)}')
+REPO_NAME=$(git config --get remote.origin.url | awk -F'[:/]' '{gsub(/\.git$/, "", $NF); print $NF}')
```
The homework assignment requires that the repository must always be named `lab0-c`. This naming convention serves as a strict identification mechanism to determine who is participating in the assignment.

jserv commented on scripts/install-git-hooks:
```diff
@@ -40,28 +41,28 @@ fi
-    *)
-      throw "Your repository MUST be forked from 'sysprog21/lab0-c'."
-  esac
+  RESPONSE=$(${CURL} -s -H "Accept: application/vnd.github.v3+json" \
```
You should ensure the content retrieved from the `curl` command is not empty.

willy-liu commented:
This change enforces that the repository name is "lab0-c" both locally 
and on GitHub while ensuring it is correctly forked from 
"sysprog21/lab0-c." It validates both the fork status and the parent 
repository name using the GitHub API.

Modifications:
- Replaced grep and sed parsing with Git commands and awk for extracting 
  the repository owner and name.
- Added validation to ensure the repository name is "lab0-c."
- Improved fork verification by checking the GitHub API response for both 
  fork status and parent repository name.

jserv reviewed (changes requested):
Refine the commit messages to avoid unnecessary bullets for modification items. Always use plain English sentences to describe.

jserv commented:
Thank @willy-liu for contributing!
